math/mathcore/src/FitUtil.cxx

```diff
@@ -337,18 +337,18 @@ double FitUtil::EvaluateChi2(const IModelFunction &func, const BinData &data, co
       // expected errors
       if (useExpErrors) {
          double invWeight  = 1.0;
+         // case of weighted Pearson chi2 fit
          if (isWeighted) {
-            // we need first to check if a weight factor needs to be applied
-            // weight = sumw2/sumw = error**2/content
-            //invWeight = y * invError * invError;
-            // we use always the global weight and not the observed one in the bin
-            // for empty bins use global weight (if it is weighted data.SumError2() is not zero)
-            invWeight = data.SumOfContent()/ data.SumOfError2();
-            //if (invError > 0) invWeight = y * invError * invError;
+            // in case of requested a weighted Pearson fit (option "PW") a weight factor needs to be applied
+            // the bin inverse weight is estimated from bin error and bin content
+            if (y != 0)
+               invWeight = y * invError * invError;
+            else
+               // when y is 0 we use a global weight estimated form all histogram (correct if scaling the histogram)
+               // note that if the data is weighted data.SumOfError2 will not be equal to zero
```

Comment:
Okay, the "correct if scaling the histogram" kind of answers my other question. But what if the histogram is not simply scaled, but all weights are different? How meaningful is the fit then if it has empty bins?